client/QuickStart/Program.cs

```diff
@@ -0,0 +1,767 @@
+    /// connecting to, administering, ingesting data into and querying Azure Data Explorer using the azure-kusto C# SDK.
+    /// You can use it as a baseline to write your own first kusto client application, altering the code as you go,
+    /// or copy code sections out of it into your app.
+    /// Tip: The app includes comments with tips on recommendations, coding best practices, links to reference materials
+    /// and recommended TO DO changes when adapting the code to your needs.
+    /// </summary>
+    public static class KustoSampleApp
+    {
+        //########################### Constants Start ###########################// 
+
+        // TODO (config - optional): Change the authentication method from "User Prompt" to any of the other options
+        //  Some of the auth modes require additional environment variables to be set in order to work
+        // (see usage in generate_connection_string below).
+        //  Managed Identity Authentication only works when running as an Azure service (webapp, function, etc.)
+        private const string
+            AuthenticationMode = "UserPrompt"; // Options: (UserPrompt|ManagedIdentity|AppKey|AppCertificate)
+
+        // TODO (config - optional): Toggle to False to execute this script "unattended"
+        private const bool WaitForUser = true;
+
+        // TODO (config):
+        // If this quickstart app was downloaded from OneClick, kusto_sample_config.json should be pre-populated
+        // with your cluster's details.
+        // If this quickstart app was downloaded from GitHub, edit kusto_sample_config.json and modify the cluster URL
+        // and database fields appropriately.
+        private const string ConfigFileName = @"kusto_sample_config.json";
+        private const int WaitForIngestSeconds = 20;
+        private static int _step = 1;
+
+        private const string BatchingPolicy = "{ 'MaximumBatchingTimeSpan': '00:00:10', 'MaximumNumberOfItems': 500, " +
+                                              "'MaximumRawDataSizeMB': 1024 }";
+        //########################### Constants End ###########################// 
+
+
+        public static async Task Main()
+        {
+            Console.WriteLine("Kusto sample app is starting...");
+            await Start();
+            Console.WriteLine("\nKusto sample app done");
+        }
+
+
+        /// <summary>
+        /// Main engine - runs the actual script. 
+        /// </summary>
+        private static async Task Start()
+        {
+            var config = LoadConfigs(ConfigFileName);
+
+            if (AuthenticationMode == "UserPrompt")
+                WaitForUserToProceed("You will be prompted *twice* for credentials during this script. " +
+                                     "Please return to the console after authenticating.");
+
+            var kustoConnectionString = GenerateConnectionString(config?.KustoUri, AuthenticationMode,
```

Comment:
You don't need to ?. syntax, you already checked that they are not null earlier.